lib/origen/generator/pattern_sequence.rb

```diff
@@ -18,9 +18,18 @@ def initialize(name, block)
 
       # Execute the given pattern
       def run(pattern_name)
-        pattern = Origen.generator.pattern_finder.find(pattern_name.to_s, {})
-        pattern = pattern[:pattern] if pattern.is_a?(Hash)
-        load pattern
+        ss "START OF PATTERN: #{pattern_name}"
+        # Give the app a chance to handle pattern dispatch
+        skip = false
+        Origen.app.listeners_for(:before_pattern_lookup).each do |listener|
+          skip ||= !listener.before_pattern_lookup(pattern_name.to_s)
+        end
+        unless skip
+          pattern = Origen.generator.pattern_finder.find(pattern_name.to_s, {})
+          pattern = pattern[:pattern] if pattern.is_a?(Hash)
+          load pattern
+        end
+        ss "END OF PATTERN: #{pattern_name}"
```

Comment:
Sorry if not as familiar with this code, but how does this change reconcile with similar code in generator.rb in run method when :sequence option is true?   I guess that code calls this? And since `skip` isn't passed then we need to check again?

Reply:
Do you have a link? I don't see a run method in generator.rb...

Reply:
Sorry-- it was generator/job.rb.   

https://github.com/Origen-SDK/origen/blob/master/lib/origen/generator/job.rb#L154

Reply:
I see.
The code in job.rb is used when looking up the top-level pattern/sequence to run. When running a pattern that is the end of it.
However, a sequence can be comprised of other patterns and sequences which are invoked via this run method.
Within the sequence you can say `seq.run :my_pattern` where the pattern name is the same as you would give to `origen g`.
The original implementation of sequences did not account for the fact that these patterns could be dispatched, so this adds the hook to handle that.

Reply:
Ok thanks-- wasn't up to speed on sequences.  I guess just make sure any other features or options used with job.rb are carried over to pattern_sequence.rb if it applies.